Add 3D operations for layers: conv3d, avg_pool3d, max_pool3d and conv3d_transpose

Kongsea opened the pull request:
1.Add 3D operations for layers:

conv3d
avg_pool3d
max_pool3d
conv3d_transpose

2.Replace a deprecated function in layers_test.py.

3.Add unit test for avg_pool3d and max_pool3d.

Kongsea committed:
Add avg_pool3d operation for layers.

Kongsea committed:
Add max_pool3d operation for layers.

Kongsea committed:
Add conv3d and convolution3d operations for layers.

Kongsea committed:
The function loss_ops.get_regularization_losses() from tensorflow.contrib.losses.python.losses has been deprecated. Use tensorflow.python.ops.losses.losses.get_regularization_losses() to replace it.

Kongsea committed:
Add unit test for avg_pool3d.

Kongsea committed:
Add unit test for max_pool3d.

Kongsea committed:
Add conv3d_transpose and convolution3d_transpose for layers.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

tensorflow-jenkins commented:
Can one of the admins verify this patch?

xcyan commented:
What's the status of this PR?

Kongsea commented:
@xcyan It's still awaiting the tensorflower's reviewing.

fchollet commented on tensorflow/contrib/layers/python/layers/layers.py:
```diff
@@ -1178,6 +1234,115 @@ def convolution2d_transpose(
+    outputs_collections=None,
+    trainable=True,
+    scope=None):
+  """Adds a convolution3d_transpose with an optional batch normalization layer.
```
Needs a line break after this line

fchollet commented on tensorflow/contrib/layers/python/layers/layers_test.py:
```diff
@@ -49,6 +48,7 @@
 from tensorflow.python.ops import variable_scope
 from tensorflow.python.ops import variables as variables_lib
 from tensorflow.python.platform import test
+from tensorflow.python.ops.losses.losses import get_regularization_losses
```
Don't import methods, only import at module level.

Kongsea committed:
Update layers.py

Add a missing line break

Kongsea committed:
Update layers_test.py

Change from importing methods to import at module level.

Kongsea commented:
Thank you. I have fixed the problems you mentioned.

fchollet reviewed (approved):
LGTM assuming tests pass.

Kongsea committed:
Update RELEASE.md

Add the description for `conv3d`, `avg_pool3d`, `max_pool3d` and `conv3d_transpose`.

Kongsea committed:
Update RELEASE.md

Update to Version 1.2

jhseu commented:
Some changes from elsewhere got into RELEASE.md. Mind reverting those?

jhseu commented:
Jenkins, test this please

Kongsea committed:
Revert "Update RELEASE.md"

This reverts commit a8ae1fedd3d3695bd0266732083d89b3af9f8ca1.

Kongsea commented:
@jhseu OK, now I have reverted the changes of RELEASE.md. Please check it.

Kongsea committed:
Revert "Update RELEASE.md"

This reverts commit 7b8c5eb6bdbd28387adf3afbfc8b49375ed0e2c8.

Kongsea committed:
Merge branch 'master' into master

Kongsea committed:
Add a missing declaration

Add missing conv3d declaration or slim.conv3d cannot be found.

martinwicke commented:
Jenkins, test this please.

Kongsea commented:
After testing, It gives "Some checks were not successful.". When I look at the testing outputs, I found an error:

Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local_linux-opt/bin/tensorflow/contrib/layers/layers_test.runfiles/org_tensorflow/tensorflow/contrib/layers/python/layers/layers_test.py", line 51, in <module>
    from tensorflow.python.ops.losses import losses
ImportError: No module named losses

However, when I run it on my computer, or import the losses module directly, it has no error. So what's the reason? What need I do to fix it?
Thank you.

martinwicke commented on tensorflow/contrib/layers/python/layers/layers_test.py:
```diff
@@ -49,6 +48,7 @@
 from tensorflow.python.ops import variable_scope
 from tensorflow.python.ops import variables as variables_lib
 from tensorflow.python.platform import test
+from tensorflow.python.ops.losses import losses
```
Can you put this in alphabetical order, i.e. switch it with the line before?

martinwicke reviewed (dismissed):
The reason for the test failure is that in contrib/layers/BUILD, layers_test does not depend on tensorflow/python/ops/losses, so when you execute the test, it isn't given access to these files. Just add that dependency there and you should be fine.

Kongsea committed:
Update layers_test.py

Put the imports in alphabetical order

Kongsea committed:
Update BUILD

Add dependency.

Kongsea commented:
@martinwicke Thank you.
Now, I have added the dependency for losses and rescheduled the order of the import. Please check it.

martinwicke commented:
Jenkins, test this please.

martinwicke commented:
It needs to be "tensorflow/python/ops/losses".

```
ERROR: /workspace/tensorflow/contrib/layers/BUILD:116:1: no such package 'tensorflow/python/losses': BUILD file not found on package path and referenced by '//tensorflow/contrib/layers:layers_test'.
ERROR: Analysis of target '//tensorflow/contrib/layers:layers_test' failed; build aborted.
```

Kongsea committed:
Update BUILD

Kongsea commented:
@martinwicke Thank you. I have modified it. Please test it again.

Kongsea committed:
Update BUILD

martinwicke commented:
Jenkins, test this please.

martinwicke commented:
It needs to be `//tensorflow/python/ops/losses` or `//tensorflow/python/ops/losses:losses`, without the _py.

Kongsea committed:
Update BUILD

Kongsea commented:
@martinwicke Sorry.
Now I have modified it. Please check and test it again.

martinwicke commented:
Jenkins, test this please!

xcyan commented:
Fixed #9347
